crates/router/src/connector/utils.rs

```diff
@@ -525,3 +525,26 @@ pub fn collect_and_sort_values_by_removing_signature(
     values.sort();
     values
 }
+
+static DENOMINATION: Lazy<HashMap<storage_models::enums::Currency, i32>> = Lazy::new(|| {
+    let mut map = HashMap::new();
+    map.insert(storage_models::enums::Currency::INR, 100);
+    map.insert(storage_models::enums::Currency::USD, 100);
+    map
+});
+
+pub fn convert_to_higher_denomination(
+    amount: i64,
+    currency: storage_models::enums::Currency,
+) -> Result<f64, error_stack::Report<errors::ConnectorError>> {
+    let factor = match DENOMINATION.get(&currency) {
+        Some(factor) => Ok(factor),
+        None => Err(errors::ConnectorError::NotImplemented(
+            "Payment Currency".to_string(),
+        )),
+    }?;
+    let amount_u32 = u32::try_from(amount)
+        .into_report()
+        .change_context(errors::ConnectorError::RequestEncodingFailed)?;
+    Ok(f64::from(amount_u32) / f64::from(*factor))
```

Comment:
```suggestion
    let amount = to_currency_base_unit(amount, currency)?;
    let amount_u32 = u32::try_from(amount)
        .into_report()
        .change_context(errors::ConnectorError::RequestEncodingFailed)?;
    Ok(f64::from(amount_u32))
```

Reply:
`u32::try_from(amount)` then redirects to mod.rs instead of `u32`
Better implementation that I was able to overcome was:
```rust
pub fn convert_to_higher_denomination(
    amount: i64,
    currency: storage_models::enums::Currency,
) -> Result<f64, error_stack::Report<errors::ConnectorError>> {
    let amount = to_currency_base_unit(amount, currency)?;
    let amount_f64: f64 = amount.parse().into_report().change_context(errors::ConnectorError::RequestEncodingFailed)?;
    Ok(amount_f64)
}
```